object_storage_buckets.go

```diff
@@ -102,6 +102,16 @@ func (c *Client) ListObjectStorageBuckets(ctx context.Context, opts *ListOptions
 	return response.Data, nil
 }
 
+// ListObjectStorageBucketsInCluster lists all ObjectStorageBuckets of a cluster
+func (c *Client) ListObjectStorageBucketsInCluster(ctx context.Context, opts *ListOptions, clusterID string) ([]ObjectStorageBucket, error) {
+	response := ObjectStorageBucketsPagedResponse{}
```

Comment:
To make sure the client is fetching buckets only for the specified cluster, a few modifications might need to be made to the `ObjectStorageBucketsPagedResponse.endpoint(args ...any)` function.

I think the best approach to this would be checking if `len(args) > 0` and appending `/{args[0]}` to the endpoint path if so, effectively adding the cluster to the path if a clusterID is specified. 

See: https://www.linode.com/docs/api/object-storage/#object-storage-buckets-in-cluster-list

Reply:
Nice, looks great now! As for testing, you can just check that the resulting bucket's cluster == the cluster given as an argument 👍

Reply:
Got it. I double checked the response in the generated fixture, and the cluster names are the same in the request url and the response: 
`    url: https://api.linode.com/v4beta/object-storage/buckets/us-east-1
    method: GET
  response:
    body: '{"page": 1, "pages": 1, "results": 1, "data": [{"hostname": "go-bucket-test-def.us-east-1.linodeobjects.com",
      "label": "go-bucket-test-def", "created": "2018-01-02T03:04:05", "cluster":
      "us-east-1", "size": 0, "objects": 0}]}'`